lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -46,34 +47,101 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
                 filter_func=possibly_scored,
                 start_node=self.location,
         ):
-            descendant = course_structure[descendant_key]
+            self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
-            if not getattr(descendant, 'has_score', False):
-                continue
+        self.all_total, self.graded_total = graders.aggregate_scores(self.scores, self.display_name, self.location)
 
+    def save(self, student, subsection, course):
+        """
+        Persist the SubsectionGrade.
+        """
+        visible_blocks = [
+            BlockRecord(unicode(location), weight, score.possible)
+            for location, (score, weight) in self.locations_to_scores.iteritems()
+        ]
+        PersistentSubsectionGrade.save_grade(
+            user_id=student.id,
+            usage_key=self.location,
+            course_version=getattr(course, 'course_version', ""),
+            subtree_edited_date=subsection.subtree_edited_on,
+            earned_all=self.all_total.earned,
+            possible_all=self.all_total.possible,
+            earned_graded=self.graded_total.earned,
+            possible_graded=self.graded_total.possible,
+            visible_blocks=visible_blocks,
+        )
+
+    def load_from_data(self, model, course_structure, scores_client, submissions_scores):
+        """
+        Load the subsection grade from the persisted model.
+        """
+        for block in model.visible_blocks.blocks:
+            persisted_values = {'weight': block.weight, 'possible': block.max_score}
+            self._compute_block_score(
+                User.objects.get(id=model.user_id),
+                block.locator,
+                course_structure,
+                scores_client,
+                submissions_scores,
+                persisted_values
+            )
+
+        self.graded_total = Score(
+            model.earned_graded,
+            model.possible_graded,
+            True,
```

Comment:
Nit: for readability, please add argument name here so we can tell for which argument this boolean is being passed.